airflow-core/src/airflow/ui/src/utils/detectLanguage.ts

```
@@ -0,0 +1,90 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Parser } from "node-sql-parser";
+import { parse as parseYaml } from "yaml";
+
+export const detectLanguage = (value: string): string => {
+  const trimmed = value.trim();
+
+  // Try to detect JSON
+  if (
+    (trimmed.startsWith("{") && trimmed.endsWith("}")) ||
+    (trimmed.startsWith("[") && trimmed.endsWith("]"))
+  ) {
```

I would remove that check, Just try parsing it if it's not valid, then it's not json

---

I think this is good to be kept. It is a clever performance optimization because this is a good pre-check before a "heavy" parser gets going... if it is a longer text body then this saves some time and makes the page faster I think.

---

yea, I agree. The pre check avoids parsing which could be helpful if these fields are large. I cleanup if you think otherwise.

---

I'm pretty sure that this is exacly what all parser will do. 

For instance `rhino` js engine (mozilla js engine written in Java), you can find that:
https://github.com/mozilla/rhino/blob/master/rhino/src/main/java/org/mozilla/javascript/json/JsonParser.java#L53

`JSON.parse` will fail early if characters are not JSON parsable.

---

@pierrejeambrun , point taken. made the changes you have requested. Please let me know if these are good

---

Okay, hope the same is with YAML... then this is a convincing argument...